FirebaseAdmin/FirebaseAdmin.IntegrationTests/FirebaseAuthTest.cs

```diff
@@ -218,13 +220,49 @@ public async Task UserLifecycle()
                 Assert.False(user.Disabled);
                 Assert.NotNull(user.UserMetaData.CreationTimestamp);
                 Assert.Null(user.UserMetaData.LastSignInTimestamp);
-                Assert.Equal(2, user.ProviderData.Length);
+                Assert.Equal(3, user.ProviderData.Length);
+                var providerIds = new HashSet<string>();
```

Comment:
HashSet has a ctor that takes an enumerable. Combining that with Select, I think you could replace lines 224-228 with:
```c#
var providerIds = new HashSet<string>(user.ProviderData.Select((providerData) => providerData.ProviderId));
```

Optional.